pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -108,19 +111,19 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 		// The object is not being deleted, so if it does not have a finalizer,
 		// then lets add the finalizer and update the object.
 		if !helper.ContainsString(instance.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName) {
-			err = r.addFinalizerAndUpdate(&request)
+			err = r.addFinalizerAndUpdate(&request, reconcileRequestId)
 		}
 
 		return reconcile.Result{}, err
 	}
 	// The object is being deleted
 	logger.V(1).Info("The VM is being marked for deletion")
-	err = r.removeFinalizerAndReleaseMac(instance, &request)
+	err = r.removeFinalizerAndReleaseMac(instance, &request, reconcileRequestId)
 	return reconcile.Result{}, err
 }
 
-func (r *ReconcilePolicy) addFinalizerAndUpdate(request *reconcile.Request) error {
-	logger := log.WithName("addFinalizerAndUpdate").WithValues("virtualMachineName", request.Name, "virtualMachineNamespace", request.Namespace)
+func (r *ReconcilePolicy) addFinalizerAndUpdate(request *reconcile.Request, RequestId int) error {
+	logger := log.WithName("addFinalizerAndUpdate").WithValues("RequestId", RequestId, "virtualMachineName", request.Name, "virtualMachineNamespace", request.Namespace)
```

Comment:
Would it make sense to pass the whole logger and just extend it with the info about finalizer? It is still the same transaction. That way finalizer does not need to know about any transaction I'd being a thing. The same for other instances where you pass the id.

Passing the whole logger wuod also make logs clearer. You would see that this finalizer function has been called as a part of the reconciliation.

Reply:
you mean that the same logger instance is passed from the reconcile function, and each time I concat the `log.WithName(<func name>).WithValues(bla bla..)` to it?

Reply:
I would pass the logger from Reconcile to addFinalizerAndUpdate as an argument. I'm not sure you can add another WithName. Maybe just WithValues("operation", "adding finalizer") or "function", "addFinalizerAndUpdate"

Reply:
what do you think about adding it with `WithName`? that way in the log name I get the entire stack path: `VirtualMachine Controller.Reconcile.addFinalizerAndUpdate...`